### ISSUE TITLE
range of valence and arousal

### ISSUE THREAD

**Issue by saeu5407** [HSE-asavchenko/face-emotion-recognition#40]
Thank you for your great model.

Is the range of valence and arousal of enot_b0_8_va_mtl.pt [-1 to 1] correct?

Based on the Affectnet dataset, it looks like [-1 to 1], but I want to know the exact range.

Thank you.

**Comment by HSE-asavchenko** [comment]
Hello! Thanks for this question. In the current implementation (see section "Multi-task: FER+Valence-Arousal" in [train_emotions-pytorch.ipynb](https://github.com/HSE-asavchenko/face-emotion-recognition/blob/main/src/affectnet/train_emotions-pytorch.ipynb)), the valence and arousal values are unbounded. I tried using tanh or similar activations, but the training was unstable. Hence, it is possible that predicted valence and arousal are out of range [-1;1] but you could simply use something like np.clip to fit them into the range [-1;1].
I should also clarify that valence and arousal were mainly used for multi-task learning to improve the quality of facial expression recognition. Hence, the quality of valence/arousal prediction may be imperfect

**Comment by saeu5407** [comment]
Thank you. Then, can I understand that the training itself was conducted between [-1, 1], but the result value can be other than that?

**Comment by HSE-asavchenko** [comment]
Yes, you're correct. However, as most of valence and arousal values in the training set have absolute value significantly lower than 1, I did not notice any cases where the prediction of my network were out of bounds [-1;1]

**Comment by saeu5407** [comment]
Thank you again for the code and explanation you provided.

**Comment by av-savchenko** [comment]
Thank you for mentioning it. You're correct, the multi-task models themselves output unnormalized logics. When I trained them, CCC loss was used, which does not require the valence and arousal to be in range [-1,1]. Please see details in section "Multi-task: FER+Valence-Arousal" at [train_emotions-pytorch.ipynb](https://github.com/sb-ai-lab/EmotiEffLib/blob/main/training_and_examples/affectnet/train_emotions-pytorch.ipynb). Speaking about arxiv paper, we described our solution for VA prediction task at ABAW challenge. We trained a simple linear model using Tensorflow Keras with activation='tanh', see details at section "Valence-Arousal" [here](https://github.com/sb-ai-lab/EmotiEffLib/blob/main/training_and_examples/ABAW/ABAW6/abaw6_affwild2.ipynb)

**Comment by ceyxasm** [comment]
@av-savchenko thank you. In your suggestion, what would be a saner handling of the VA outputs? 
* Clipping and normalizing, OR
* Passing through a sigmoid?
